lib/devise/models/recoverable.rb

```diff
@@ -33,8 +33,8 @@ def self.required_fields(klass)
       # Update password saving the record and clearing token. Returns true if
       # the passwords are valid and the record was saved, false otherwise.
       def reset_password(new_password, new_password_confirmation)
-        self.password = new_password
-        self.password_confirmation = new_password_confirmation
+        self.password = new_password.to_s
```

Comment:
Instead of assigning an empty `, I think we should explicitly check for `nil` values on `new_password` and add an error to the object instead.